src/meta/src/manager/catalog.rs

```diff
@@ -319,19 +331,20 @@ where
             }
             Ok(())
         } else {
-            bail!("table already exist or not in creating procedure",)
+            Err(anyhow!("table already exist or not in creating procedure").into())
         }
     }
 
-    pub async fn drop_table(&self, table_id: TableId) -> Result<NotificationVersion> {
+    pub async fn drop_table(&self, table_id: TableId) -> MetaResult<NotificationVersion> {
         let mut core = self.core.lock().await;
         let table = Table::select(self.env.meta_store(), &table_id).await?;
         if let Some(table) = table {
             match core.get_ref_count(table_id) {
-                Some(ref_count) => Err(PermissionDenied(format!(
```

Comment:
Suggest to add a `PermissionDenied` error for `MetaErrorInner`, rather than using anyhow here.